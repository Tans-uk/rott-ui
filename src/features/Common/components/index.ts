export * from './CommonItem'
export * from './CommonItemContainer'
