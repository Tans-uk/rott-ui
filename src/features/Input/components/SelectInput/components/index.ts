export * from './SelectInput'
export * from './SelectInputModalComponent'
