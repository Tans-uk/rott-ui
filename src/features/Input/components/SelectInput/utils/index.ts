export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
export * from './sortListBySearchPriority'
