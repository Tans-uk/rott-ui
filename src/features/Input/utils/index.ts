export * from './formatByDateMode'
export * from './useInputStyleNormalizer'
