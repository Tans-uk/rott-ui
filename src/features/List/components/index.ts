export * from './List'
export * from './FavoriteSwipeComponent'
export * from './ListSkeletonItem'
