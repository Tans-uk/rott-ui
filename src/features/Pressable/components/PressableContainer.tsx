import {type FC, type PropsWithChildren} from 'react'

import {Animated, Pressable as RNPressable} from 'react-native'

import {ThemeConfig} from '../../../models'
import type {PressableProps} from '../models'

export const PressableContainer: FC<PropsWithChildren<PressableProps<ThemeConfig>>> = ({
  children: containerChildren,
  ...containerProps
}) => {
  if (containerProps.animated) {
    const AnimatedPressable = Animated.createAnimatedComponent(RNPressable)

    return <AnimatedPressable {...containerProps}>{containerChildren}</AnimatedPressable>
  } else return <RNPressable {...containerProps}>{containerChildren}</RNPressable>
}
