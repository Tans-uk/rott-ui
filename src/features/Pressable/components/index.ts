export * from './Pressable'
export * from './PressableContainer'
