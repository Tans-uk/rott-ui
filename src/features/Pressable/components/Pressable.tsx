import {isValidElement, type FC} from 'react'

import {StyleSheet} from 'react-native'

import {Label} from '../../Label'
import type {PressableProps} from '../models'
import {PressableStyles} from '../styles'
import {PressableContainer} from './PressableContainer'

export const Pressable: FC<PressableProps> = ({
  text,
  textStyle,
  textVariant = 'grey-900',
  textSize = 'md',
  textWeight,
  animated = false,
  children,
  style,
  ref,
  ...props
}) => {
  return (
    <PressableContainer
      animated={animated}
      ref={ref}
      testID='pressable-test-id'
      style={StyleSheet.flatten([
        PressableStyles({
          includeAlignItems: true,
          includeJustifyContent: true,
          ...props,
        }).defaultPressableStyle,
        style,
      ])}
      {...props}>
      {text && !isValidElement(text) && (
        <Label style={textStyle} variant={textVariant} fontSize={textSize} fontWeight={textWeight}>
          {text}
        </Label>
      )}

      {!text && <>{children}</>}
    </PressableContainer>
  )
}
