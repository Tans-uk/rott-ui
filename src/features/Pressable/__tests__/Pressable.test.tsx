import {Pressable} from '../components'

import {commonUiTestExtension} from 'src/__tests__/utils/commonUiTestExtension'
import {fireEvent, render} from 'src/__tests__/utils/testUtils'

const testId = {
  pressableTestId: 'pressable-test-id',
  pressableLabel: 'Pressable Label',
}

describe('Pressable -> Custom Component', () => {
  it('pressable component render olmalı ve snapshot eşleşmeli', () => {
    const rendered = render(<Pressable />)

    expect(rendered).toMatchSnapshot()
  })

  it('pressable componenti label ile render olmalı', () => {
    const {pressableTestId, pressableLabel} = testId
    const {getByTestId, getByText} = render(
      <Pressable text={pressableLabel} testID={pressableTestId} />
    )

    const pressableElement = getByTestId(pressableTestId)
    const labelElement = getByText(pressableLabel)

    expect(pressableElement).toBeTruthy()
    expect(labelElement).toBeTruthy()
  })

  it('pressable tıklanabilir olmalı', () => {
    const {pressableTestId, pressableLabel} = testId
    const onPressMock = jest.fn()
    const {getByTestId} = render(
      <Pressable
        text={pressableLabel}
        textVariant='black'
        textSize='md'
        textWeight='bold'
        testID={pressableTestId}
        onPress={onPressMock}
      />
    )
    const pressableElement = getByTestId(pressableTestId)
    expect(pressableElement).toBeTruthy()

    fireEvent.press(pressableElement)
    expect(onPressMock).toHaveBeenCalled()
  })

  it('pressable disableken tıklanmamalı', () => {
    const {pressableTestId, pressableLabel} = testId
    const onPressMock = jest.fn()
    const {getByTestId} = render(
      <Pressable
        text={pressableLabel}
        textVariant='black'
        textSize='md'
        textWeight='bold'
        testID={pressableTestId}
        onPress={onPressMock}
        disabled
      />
    )

    const pressableElement = getByTestId(pressableTestId)
    expect(pressableElement).toBeTruthy()

    fireEvent.press(pressableElement)
    expect(onPressMock).not.toBeCalled()
  })

  it('pressable componenti style ile render olmalı', () => {
    const {pressableTestId, pressableLabel} = testId
    const onPressMock = jest.fn()
    const {getByText} = render(
      <Pressable
        text={pressableLabel}
        textVariant='black'
        textSize='md'
        textWeight='bold'
        testID={pressableTestId}
        onPress={onPressMock}
      />
    )

    const textElement = getByText(pressableLabel)

    expect(textElement).toHaveStyle({fontWeight: 'bold'})
  })

  commonUiTestExtension(<Pressable testID={testId.pressableTestId} />, testId.pressableTestId)
})
