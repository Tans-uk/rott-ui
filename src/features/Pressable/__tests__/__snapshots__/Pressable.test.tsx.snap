// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pressable -> Custom Component pressable component render olmalı ve snapshot eşleşmeli 1`] = `
<View
  flex={1}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animated={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flex": 1,
        "flexDirection": "column",
      }
    }
    testID="pressable-test-id"
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>
</View>
`;
